pkg/cli/userfiletable_test.go

```diff
@@ -80,6 +81,156 @@ func Example_userfile_upload() {
 	// successfully uploaded to userfile://defaultdb.public.userfiles_root/test/file3.csv
 }
 
+func Example_userfile_upload_recursive() {
+	if err := runExampleUserfileUploadRecursive(); err != nil {
+		fmt.Println(err)
+	}
+}
+
+func runExampleUserfileUploadRecursive() (err error) {
+	c := NewCLITest(TestCLIParams{})
+	defer c.Cleanup()
+
+	tmpDir, err := ioutil.TempDir("", testUserfileUploadTempDirPrefix)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = errors.CombineErrors(err, os.RemoveAll(tmpDir))
+	}()
+	testDir := filepath.Join(tmpDir, "testdir/")
+	if err = os.MkdirAll(filepath.Join(testDir, "d1/d11/d111/d1111"), 0755); err != nil {
+		return err
+	}
+	if err = os.MkdirAll(filepath.Join(testDir, "d2/d21"), 0755); err != nil {
+		return err
+	}
+
+	filesRelativePaths := []string{
+		"d1/" + testTempFilePrefix + "f1.csv",
+		"d1/d11/" + testTempFilePrefix + "f2.csv",
+		"d1/d11/" + testTempFilePrefix + "f3.csv",
+		"d1/d11/d111/d1111/" + testTempFilePrefix + "f4.csv",
+		"d2/d21/" + testTempFilePrefix + "f5.csv",
+	}
+	for _, relPath := range filesRelativePaths {
+		if err := ioutil.WriteFile(filepath.Join(testDir, relPath), []byte("content"), 0666); err != nil {
+			return err
+		}
+	}
+
+	c.RunWithArgs([]string{"userfile", "upload", "-r", testDir, "foo"})
```

Comment:
nice tests! Do we have a test where we exercise the case where srcDir does not have a trailing slash. I think `testDir` always has one, so wondering if we should add a case where we have testDir but without the `/`.